telethon/utils.py

```diff
@@ -1101,104 +1103,85 @@ def resolve_bot_file_id(file_id):
-            # By checking the `dc_id`, we greatly reduce the chances of this
-            # happening.
-            return None
-
+    if isinstance(file, DocumentFileId):
         attributes = []
-        if file_type == 3 or file_type == 9:
+        if file.type_id in (FileId.TYPE_VOICE, FileId.TYPE_AUDIO):
             attributes.append(types.DocumentAttributeAudio(
                 duration=0,
-                voice=file_type == 3
+                voice=file.type_id == FileId.TYPE_VOICE
             ))
-        elif file_type == 4 or file_type == 13:
+        elif file.type_id == FileId.TYPE_VIDEO or file.type_id == FileId.TYPE_VIDEO_NOTE:
             attributes.append(types.DocumentAttributeVideo(
                 duration=0,
                 w=0,
                 h=0,
-                round_message=file_type == 13
+                round_message=file.type_id == FileId.TYPE_VIDEO_NOTE
             ))
-        # elif file_type == 5:  # other, cannot know which
-        elif file_type == 8:
+        # elif file.type_id == FieldId.TYPE_DOCUMENT:  # other, cannot know which --> no attribute
+        elif file.type_id == FileId.TYPE_STICKER:
             attributes.append(types.DocumentAttributeSticker(
                 alt='',
                 stickerset=types.InputStickerSetEmpty()
             ))
-        elif file_type == 10:
+        elif file.type_id == FileId.TYPE_ANIMATION:
             attributes.append(types.DocumentAttributeAnimated())
+        # end if
 
         return types.Document(
-            id=media_id,
-            access_hash=access_hash,
+            id=file.id,
+            access_hash=file.access_hash,
             date=None,
             mime_type='',
             size=0,
             thumbs=None,
-            dc_id=dc_id,
+            dc_id=file.dc_id,
             attributes=attributes,
-            file_reference=b''
+            file_reference=file.file_reference if file.has_reference else b''
         )
-    elif (version == 2 and len(data) == 44) or (version == 4 and len(data) in (49, 77)):
-        if version == 2:
-            (file_type, dc_id, media_id, access_hash,
-                volume_id, secret, local_id) = struct.unpack('<iiqqqqi', data)
-        # else version == 4:
-        elif len(data) == 49:
-            # TODO Figure out what the extra five bytes mean
-            (file_type, dc_id, media_id, access_hash,
-                volume_id, secret, local_id, _) = struct.unpack('<iiqqqqi5s', data)
-        elif len(data) == 77:
-            # See #1613.
-            (file_type, dc_id, _, media_id, access_hash, volume_id, _, local_id, _) = struct.unpack('<ii28sqqq12sib', data)
-        else:
-            return None
+    elif not isinstance(file, PhotoFileId):  # e.g. WebLocationFileId
+        return None
+    # end if
 
-        if not (1 <= dc_id <= 5):
-            return None
+    # okey, so we're dealing with a photo.
+    assert isinstance(file, PhotoFileId)
```

Comment:
I think I'd rather have this `return None` if it's not a photo:

```suggestion
    if not isinstance(file, PhotoFileId):
        return None
```